app/templates/courses/teacher-classes-view.jade

```diff
@@ -127,3 +131,70 @@ mixin archivedClassRow(classroom)
     .col-xs-2
       .class-links.pull-right
         a.unarchive-classroom.text-h6(data-i18n='teacher.unarchive_class' data-classroom-id=classroom.id)
+
+mixin teacher-quests
+  - var numComplete = 0;
+  - var totalQuests = 0;
+  each quest in view.teacherQuestData
+    - totalQuests++;
+    if quest.complete
+      - lastComplete = quest;
+      - numComplete++;
+    else if !nextIncomplete
+      - nextIncomplete = quest;
+  if lastComplete
+    - lastComplete.lastComplete = true;
+  if nextIncomplete
+    - nextIncomplete.nextIncomplete = true;
+  - var percentComplete = totalQuests ? Math.round(numComplete * 100.0 / totalQuests) : 0;
+  if view.prepaids.length < 1
+    .teacher-quests
+      .row
+        .col-sm-3.quests-summary-col
+          .quest-progress.center #{percentComplete}%
+          .center
+            if numComplete != totalQuests
+              .el
+                - var height = 100;
+                svg(width=height, height=height)
+                  - var radius = height / 2;
+                  - var fullCircleStroke = 2*Math.PI*radius / 2
+                  circle.bottom(r=radius,cx=radius,cy=radius)
+                  circle.top(r=radius / 2,cx=radius,cy=radius,style="stroke-dasharray: #{fullCircleStroke/100*percentComplete} #{fullCircleStroke}")
+          .center Quests Complete
+        .col-sm-9.quests-list-col
+          if numComplete != totalQuests
+            h3 Teacher Quest for Success
+            strong.center!= view.translateWithMarkdown('teacher.teacher_quest_keep_going')
+          else
+            strong.center!= view.translateWithMarkdown('teacher.teacher_quest_done')
+          - var lastComplete = null;
+          - var nextIncomplete = null;
```

Comment:
In compact mode we show the first incomplete item on the list, and also the complete one thats right before it.